Automattic/jetpack
# Classic Theme Helper: Featured content remove wpcom only code

Related to https://github.com/Automattic/vulcan/issues/348

## Proposed changes:

As explained in https://github.com/Automattic/jetpack/pull/37260#discussion_r1592423312. The hooks for featured content that only pertains to simple should be moved to the featured-content mu-plugin in wpcom.

### Other information:


## Jetpack product discussion


pfwV0U-6Y-p2
## Does this pull request change what data or activity we track or use?


No
## Testing instructions:

Copying the amazing instructions provided in https://github.com/Automattic/jetpack/pull/37969 by @coder-karen 
**Self-hosted / Atomic / Simple:**

To test existing behaviour:
* With a theme that supports Featured Content (Dara is an example), open the Customizer. You should see a 'Featured Content' section. When opening the page source from here, you should see the `featured-content-suggest-js` (and that it's source on Atomic is from the `jetpack-mu-wpcom-plugin` plugin (or `jetpack-mu-wpcom-plugin-dev` if using the beta tester plugin), or `jetpack` / `jetpack-dev` on self-hosted sites ). 
* Confirm everything works as it should: Add a tag and click save, then apply that tag to a few posts - the featured content should display on the homepage.

To test this PR:
* For responsive videos, follow the steps above, then view the post / page source for the published post / page. Notice the `jetpack-responsive-videos-js` script in the page source (and that it's source is from the `jetpack-mu-wpcom-plugin` on Atomic and `jetpack` on self-hosted sites ) , and that the video is wrapped in a div with the class `jetpack-video-wrapper`.
* For the Featured Content changes in this PR, follow the same testing steps as when testing existing behaviour.
* Everything should test as it did before (and the `featured-content-suggest-js` script source is the same as before as well).
